mailet/test/src/main/java/org/apache/mailet/base/test/FakeMail.java

```diff
@@ -172,12 +172,22 @@ public Builder recipient(String recipient) throws AddressException {
         public Builder name(String name) {
             Preconditions.checkNotNull("'name' can not be null");
             this.name = Optional.of(name);
+            return this.name(Optional.of(name));
+        }
+
+        public Builder name(Optional<String> name) {
```

Comment:
two `name()` methods are not consistent. One doesn't accept null value, one accept empty Optional? 
```java
        public Builder name(String name) {
            Preconditions.checkNotNull("'name' can not be null");
            this.name = Optional.of(name);
            return this;
        }

        public Builder name(Optional<String> name) {
            this.name = name;
            return this;
        }
```
and similar to other methods.